src/Core/Manifest/Module.php

```diff
@@ -210,7 +210,7 @@ public function __unserialize(array $data): void
      */
     public function serialize()
     {
-        return json_encode([$this->path, $this->basePath, $this->composerData]);
+        return json_encode($this->__serialize());
```

Comment:
The Serializable interface wants to you to use `__serialize` and `__unserialize` now.

Reply:
I think this is incorrect, as `__serialize()` returns an assoc array, while the old `json_encode` approach uses a regular array.

There's enough smarts in PHP to handle BC, so during at unserialize time to detect if the data is in an assoc array format or other, and then it will call either __unserialize() or unserialize as required.